tests/tests/wgpu-validation/util.rs

```diff
@@ -39,3 +39,66 @@ fn staging_belt_random_test() {
         belt.recall();
     }
 }
+
+#[test]
+fn staging_belt_panics_with_invalid_buffer_usages() {
+    #[track_caller]
+    fn test_if_panics(usage: wgpu::BufferUsages) {
+        if let Err(panic) = std::panic::catch_unwind(|| {
+            let (device, _queue) = wgpu::Device::noop(&wgpu::DeviceDescriptor::default());
+            let _belt = wgpu::util::StagingBelt::new_with_buffer_usages(device.clone(), 512, usage);
+        }) {
+            // according to [1] the panic payload is either a `&str` or `String`
+            // [1]: https://doc.rust-lang.org/std/macro.panic.html
+
+            let message = if let Some(message) = panic.downcast_ref::<&str>() {
+                *message
+            } else if let Some(message) = panic.downcast_ref::<String>() {
+                message.as_str()
+            } else {
+                // don't know what this panic is, but it's not ours
+                std::panic::resume_unwind(panic);
+            };
+
+            let expected_message = format!("Only BufferUsages::COPY_SRC may be used when Features::MAPPABLE_PRIMARY_BUFFERS is not enabled. Specified buffer usages: {usage:?}");
+            if expected_message == message {
+                // panicked with the correct message
+            } else {
+                // This is not our panic (or the panic message was changed)
+                std::panic::resume_unwind(panic);
+            }
+        } else {
+            panic!("StagingBelt::new_with_buffer_usages should panic without MAPPABLE_PRIMARY_BUFFERS with usage={usage:?}");
+        }
+    }
+
+    for mut usage in wgpu::BufferUsages::all()
+        .difference(wgpu::BufferUsages::COPY_SRC | wgpu::BufferUsages::MAP_WRITE)
+        .iter()
+    {
+        test_if_panics(usage);
+
+        usage.insert(wgpu::BufferUsages::MAP_WRITE);
+        test_if_panics(usage);
+    }
+}
+
+#[test]
+fn staging_belt_works_with_non_exclusive_buffer_usages() {
```

Comment:
We might consider a test for Mappable Primary Buffers too - the noop device supports all features and all limits.